velero-plugin-for-aws/restoreplugin.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2017, 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	corev1api "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	envAZOverride = "VELERO_AWS_AZ_OVERRIDE"
+	labelZoneBeta = "failure-domain.beta.kubernetes.io/zone"
+	labelZone     = "topology.kubernetes.io/zone"
+)
+
+// RestorePlugin is a restore item action plugin for Velero
+type RestorePlugin struct {
+	log logrus.FieldLogger
+}
+
+func newRestorePlugin(logger logrus.FieldLogger) *RestorePlugin {
+	return &RestorePlugin{log: logger}
+}
+
+// AppliesTo returns information about which resources this action should be invoked for.
+// A RestoreItemAction's Execute function will only be invoked on items that match the returned
+// selector. A zero-valued ResourceSelector matches all resources.g
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"persistentvolume"},
```

Comment:
```suggestion
		IncludedResources: []string{"persistentvolumes"},
```

I think? Though even with that change, I cannot get `Execute` to run (or at least to print its initial log message) even though I confirm that `AppliesTo` ran. Still digging into why.

Reply:
Found it: `Execute` (in the restore plugin) is skipped when `CreateVolumeFromSnapshot` (in the snapshotter) fails.